metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_schema.py

```diff
@@ -8,49 +8,33 @@
 from google.cloud.bigquery.table import RowIterator, TableListItem, TimePartitioning
 
 from datahub.ingestion.source.bigquery_v2.bigquery_audit import BigqueryTableIdentifier
+from datahub.ingestion.source.sql.sql_generic import BaseColumn, BaseTable, BaseView
 
 logger: logging.Logger = logging.getLogger(__name__)
 
 
 @dataclass(frozen=True, eq=True)
-class BigqueryColumn:
-    name: str
-    ordinal_position: int
+class BigqueryColumn(BaseColumn):
     field_path: str
-    is_nullable: bool
     is_partition_column: bool
-    data_type: str
-    comment: str
 
 
 @dataclass
-class BigqueryTable:
-    name: str
-    created: datetime
-    last_altered: Optional[datetime]
-    size_in_bytes: Optional[int]
-    rows_count: Optional[int]
-    expires: Optional[datetime]
-    clustering_fields: Optional[List[str]]
-    labels: Optional[Dict[str, str]]
-    num_partitions: Optional[int]
-    max_partition_id: Optional[str]
-    max_shard_id: Optional[str]
-    active_billable_bytes: Optional[int]
-    long_term_billable_bytes: Optional[int]
-    comment: str
-    ddl: str
-    time_partitioning: TimePartitioning
+class BigqueryTable(BaseTable):
+    expires: Optional[datetime] = None
+    clustering_fields: Optional[List[str]] = None
+    labels: Optional[Dict[str, str]] = None
+    num_partitions: Optional[int] = None
+    max_partition_id: Optional[str] = None
+    max_shard_id: Optional[str] = None
+    active_billable_bytes: Optional[int] = None
+    long_term_billable_bytes: Optional[int] = None
+    time_partitioning: Optional[TimePartitioning] = None
     columns: List[BigqueryColumn] = field(default_factory=list)
 
 
 @dataclass
-class BigqueryView:
-    name: str
-    created: datetime
-    last_altered: datetime
-    comment: str
-    ddl: str
+class BigqueryView(BaseView):
```

Comment:
For 100% correctness, this can be changed and the columns field can be removed. However, it's probably good enough as-is so no need to change it

```suggestion
class BigqueryView(BaseView[BigqueryColumn]):
```